### ISSUE TITLE
Mic and Hold activating automatically when returning to the first call

### ISSUE THREAD

**Issue by EmbitServ** [InnovateAsterisk/Browser-Phone#342]
Hello Conrad,

Two situations are occurring... when I'm on 2 or more calls, muting the mic or holding works normally on the call that is running. But when I jump to another call (and put the first one on hold) the mic activates again (hold also deactives) and the mic icon keeps saying it is muted, but the mic has been activated.

The same goes for Hold. When I press hold on the call it activates it, but when I return to the call the hold is deactivated.

Basically what is happening is that when returning to call waiting, both the microphone is being activated and the hold is being deactivated automatically.

**Comment by InnovateAsterisk** [comment]
> Basically what is happening is that when returning to call waiting, both the microphone is being activated and the hold is being deactivated automatically.

Yes, correct, this is the intended behaviour.
The mic mute is important behaviour that I implemented.

Previously, when you put the call on hold, your mic is actually active AND TRANSMITTING! technically a hold in the SDP is set as ```sendonly``` - this may not sound like you are "_putting the call on hold_", but its actually correct, what should happen is by setting the SDP to ```sendonly``` you are actually saying "_i dont want to hear what the other person is saying_" so this is why often music is playing to the B at this point. Systems like Asterisk don't transmit on your audio to the B side even though it may be transmitting, they play music.

Now the problem is, if you stop the stream, Asterisk could see this as no-audio and kill the call. Even though asterisk has a special hold timer, I found it better to just mute the audio (send a 0 stream), keeping the flow of packets in at least one direction. (remember... `send only` - as in we will be sending audio only, so if you set that, but then don't send any packets, you could be in trouble)

Also, what I found is that during a conference call, pressing hold stops you from hearing the conversation (as you would expect), but the other participants CAN HEAR YOU! Remember the instruction is ```sendonly```, and because its a conference, Asterisk cant suddenly mask your audio with hold music... lol that would be rather annoying for the other parties.

Either way, hold is best done with ```sendonly``` AND a mic mute. I think the only problem here is that If you had set the mic to mute before switching calls... it may have restored the call and the mic, but not updated the UI.
